bigtable/tests/admin_integration_test.cc

```diff
@@ -12,185 +12,269 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <gmock/gmock.h>
+
+#include <absl/memory/memory.h>
 #include <absl/strings/str_join.h>
 #include <google/protobuf/text_format.h>
 #include <google/protobuf/util/message_differencer.h>
-#include <sstream>
-#include "bigtable/admin/admin_client.h"
-#include "bigtable/admin/table_admin.h"
 
-namespace {
+#include <bigtable/admin/admin_client.h>
+#include <bigtable/admin/table_admin.h>
+#include "bigtable/client/testing/table_integration_test.h"
+
+#include <string>
+#include <vector>
+
+using namespace std;
 namespace admin_proto = ::google::bigtable::admin::v2;
 
-void CheckInstanceIsEmpty(bigtable::TableAdmin& admin) {
-  std::vector<std::string> table_names;
-  auto table_list = admin.ListTables(admin_proto::Table::NAME_ONLY);
-  if (not table_list.empty()) {
-    throw std::runtime_error("Expected empty instance in integration test");
-  }
-  std::cout << "Initial ListTables() successful" << std::endl;
-}
+namespace bigtable {
+namespace testing {
+
+class AdminIntegrationTest : public TableIntegrationTest {
+ protected:
+  std::unique_ptr<bigtable::TableAdmin> table_admin_;
 
-void CheckCreateTable(bigtable::TableAdmin& admin, std::string table_id) {
-  auto table = admin.CreateTable(table_id, bigtable::TableConfig());
-  std::cout << "CreateTable(" << table_id << ") successful" << std::endl;
+  void SetUp() { CreateTableAdmin(); }
 
-  auto get_result = admin.GetTable(table_id);
-  if (table.name() != get_result.name()) {
-    std::ostringstream os;
-    os << "Mismatched names for GetTable(" << table_id << "): " << table.name()
-       << " != " << get_result.name();
-    throw std::runtime_error(os.str());
+  void TearDown() {}
+
+  void CreateTableAdmin() {
```

Comment:
This function is only called from `SetUp()`, why not move the code there and remove the function?